### ISSUE TITLE
An identifier can't start with a dollar sign anymore

### ISSUE THREAD

**Issue by JustusGreiberORGADATA** [sebastienros/fluid#496]
I just tried to update Fluid from 2.2.0 to the newest version (yes that is quite the jump 😃) and noticed that after #423 was merged, I can't use a dollar sign to start my variables anymore - only numbers, letters and underscores work [0]. What I also noticed is, that any kind of unicode letters like `ä ö ü` would not work either (but I don't use them so it's not like I _really_ care, I just found it interesting as a german 😄).

I can't find any specification what kind of variable names _should_ be allowed in liquid. If this is a restriction of the language I would just change "special variables" in my templates from `$Subject` to `__Subject`. But if it's not part of the language I would lobby to bring the dollar sign back, because it's easier to spot next to all the other "vertical lines" my templates use to ignore most of the whitespace (e. g. `{-case __Subject -}`).

Cheers

[0] https://github.com/sebastienros/fluid/blob/f61b88fda54eabf606b944c94d213a9e2b3fe4ca/Fluid/Parser/IdentifierParser.cs#L89

**Comment by sebastienros** [comment]
This is the rule in the official Liquid parser: https://github.com/Shopify/liquid/blob/eab13a07d9861a38d993d2749ae25f06ff76426b/lib/liquid/lexer.rb#L1

However I am quite certain an identifier can start with a digit, which is not in this regex.

**Comment by JustusGreiberORGADATA** [comment]
Hey thanks for the quick answer 🍾. If it's in the official syntax, I will change my variable names 👍. If the offical parser doesn't allow it, it's probably better anyway, to stay compatible with most other implemenations.

> However I am quite certain an identifier can start with a digit, which is not in this regex.

In your implementation this is true, yes :). It was requested in #422. I don't really know if this is good or bad, that's for you to decide 🙃.

Anyway have a great weekend.
